spring-integration-test/src/main/java/org/springframework/integration/test/mock/MockMessageHandler.java

```diff
@@ -127,6 +130,7 @@ protected void handleMessageInternal(Message<?> message) {
 			this.lock.unlock();
 		}
 
+		Assert.notNull(function, "function must not be null");
```

Comment:
Good. Let's also add something like: 
"At least one of  the 'handleNext' or 'handleNextAndReply' has to be attached to this MockMessageHandler."